### DISCUSSION TITLE
Typescript not picking up type errors

### DISCUSSION THREAD

**Discussion by liamoreilly** [quasarframework/quasar#13808] [General - Components / Directives / etc]
Hi,

It seems to me that typescript is not picking up type errors out of the box.

I have created a new Quasar project using `npm init quasar` and the following relevant options:

- Quasar CLI
- Quasar v2 (Vue 3)
- Typescript
- Quasar App CLI with Vite
- Composition API

Then I add the following lines to `router/routes.ts`:

```
const x:string = 7;
console.log(x);
```

I would now expect to get an error when running `quasar dev` (something similar to the following when using `npx tsc`):
```
error TS2322: Type 'number' is not assignable to type 'string'.
```

Instead, the typescript compilation to javascript proceeds with no errors.

Am I missing something here?
Thanks

**Comment by metalsadman** [comment]
https://vitejs.dev/guide/features.html#typescript states that
![Screen Shot 2022-06-29 at 11 06 16 AM](https://user-images.githubusercontent.com/31038838/176342585-590088af-1e8f-4469-bf85-19111fae2b00.png)

try https://github.com/vitejs/awesome-vite
https://github.com/fi3ework/vite-plugin-checker

**Reply by liamoreilly** [reply]
Thank you for this. I am not sure how I missed this. I had looked through the docs fairly thoroughly. Most of the typescript search results were under the webpack area, but still, I am not sure how I missed this.

Thanks, this was helpful.
